express-crud/server.js

```diff
@@ -0,0 +1,143 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib';
+
+const app = express();
+
+app.use(express.json());
+
+const db = new pg.Pool();
+
+// First GET
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+      from "grades"
+    `;
+    const result = await db.query(sql);
+    const grades = result.rows;
+    if (!grades) {
+      return res.status(500).json({ error: 'Query failed' });
```

Comment:
Don't ever return a 400 or 500 status yourself. Instead, throw an exception. To throw a regular error (not a ClientError) you can use `throw new Error('message');`.

But this isn't necessary. If the query fails, `db.query` will throw an exception for you.